### ISSUE TITLE
Frame.from_xlsx produces a 1048574 row frame from this 6 row excel file

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#146]
[bad_sheet.xlsx](https://github.com/InvestmentSystems/static-frame/files/3985678/bad_sheet.xlsx)

`sf.Frame.from_xlsx('/tmp/bad_sheet.xlsx')` results in a huge sheet full of `None`s. If you try to set an index with `index_depth=1` (as happens by default in a `Bus`), Static Frame is unable to load the sheet at all because of the nonunique values in the first column.

**Comment by flexatone** [comment]
Thanks for isolating this issue!

**Comment by flexatone** [comment]
Can you let tell me how this XSLX was produced? The `openpyxl` library, we which we use for reading this XLSX file, identifiers the dimensions as for this sheet as follows:
```python
>>> ws.calculate_dimension()                                                            
'A1:F1048576'
```
Thus, there is no easy fix on our side I see at this time. Of course, you can import this with `index_depth=1` when using `Frame.from_xlsx`; I will expose setting these parameters on the `Bus`.

**Comment by ForeverWintr** [comment]
I produced it by hand in libreoffice. A couple of things that may or may not be relevant:

- There was never any data below row 6. 
- I applied a format to an entire column.
  - I tried using the `values_only` flag in `ws.iter_rows()`, thinking the format might be the culprit, but didn't see any change.
- pandas reads this sheet correctly.

Regarding setting index_depth, do you mean we can import this with index_depth=0, then drop the empty rows?

Even though there is no easy fix, I figure having this issue and the attached example is worthwhile. Known issues with openpyxl will probably play into future decisions when using the library.

**Comment by ForeverWintr** [comment]
Actually, I think this openpyxl issue may be related: https://bitbucket.org/openpyxl/openpyxl/issues/514/cell-max_row-reports-higher-than-actual

**Comment by flexatone** [comment]
Agreed. I will look into using something like this approach: 

```python
for row in reversed(ws.rows):
    values = [cell.value for cell in row]
    if any(values):
        print("last row with data is {0}".format(row[0].row))
        break
```

**Comment by flexatone** [comment]
While I was not able to use the approach shown in that issue (reverse iterating `ws.rows`, as we open the workbook with `read_only=True`), I was able to efficiently process the data after load and identify all-empty rows. Further, this approach will not affect the performance of sheets that do not have these all-empty rows.

This change will be released as part of 0.5.9.
